src/common/database-service/DuckDBClient.ts

```diff
@@ -68,4 +83,8 @@ export class DuckDBClient {
       });
     });
   }
+
+  private appendToFile() {
```

Comment:
nit: rename appendProfileToFile to make it more explicit that we are appending profile.